core/src/Kokkos_HostSpace.hpp

```diff
@@ -279,10 +279,13 @@ namespace Impl {
 
 template <class DT, class... DP>
 struct ZeroMemset<typename HostSpace::execution_space, DT, DP...> {
-  ZeroMemset(const typename HostSpace::execution_space&,
+  ZeroMemset(const typename HostSpace::execution_space& exec,
              const View<DT, DP...>& dst,
-             typename View<DT, DP...>::const_value_type& value)
-      : ZeroMemset(dst, value) {}
+             typename View<DT, DP...>::const_value_type&) {
+    hostspace_fence(exec);
```

Comment:
Why not just `exec.fence()`?

Reply:
No that's fine but add a comment in the code

Reply:
I added a comment.

Reply:
Thank you